### ISSUE TITLE
Unexpected SyntaxError in `ORDER BY`

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13091]
When running the following query against an empty database:

```cypher
RETURN null ORDER BY properties(CASE WHEN null THEN null END)
```

I encountered the following error:
```
Type mismatch: expected Map, Node or Relationship but was Boolean (line 1, column 33 (offset: 32))
"RETURN null ORDER BY properties(CASE WHEN null THEN null END)"
                                 ^
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.
Additionally, alternatives that use the same syntax constructs or evaluate to the same query don't throw an error, like for example:
`RETURN properties(CASE WHEN null THEN null END)` or
`RETURN null ORDER BY properties(CASE WHEN false THEN null END)` or
`RETURN null ORDER BY properties(null)`

I encountered this issue when testing queries against the **Neo4j 5.5.0 community version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following query and observe it throws an error:
```cypher
RETURN null ORDER BY properties(CASE WHEN null THEN null END)
```

### Expected behavior
The query should run successfully

### Actual behavior
The query fails with the error message `Type mismatch: expected Map, Node or Relationship but was Boolean...`

**Comment by HannesSandberg** [comment]
Thank you! We will investigate this.

**Comment by GauravScripts** [comment]
Yes, Same Issue encounter by me also. Please resolve the issue.

**Comment by sherfert** [comment]
The fix will be part of the upcoming neo4j 5.13 release.
